### PR TITLE
chore: add Doctrine migrations

### PR DESCRIPTION
**Pull request by tomudding**
# Description

Adds support for migrations while using Doctrine ORM. The initial migration is based on the current `main`.

Normally, a `migrations:rollup` would be used to ensure that this migration is successful. However, in our case it is necessary to export ALL data (but not structure) from the database, drop everything in the database, and only then perform the migration. Data can be easily exported from PostgreSQL using:

```sh
pg_dump -U gewisdb --column-inserts --data-only gewisdb
pg_dump -U gewisdb --column-inserts --data-only gewisdb_report
```

This ensures that _all_ FKs are the same for newer migrations preventing headaches in the future.

---

Convenience methods have been added for `make` to make using the migrations easier. However, in production everything must go directly through `./orm`.

---

Also added the proper tooling to ensure that each database connection is made using the `SET ROLE` query.

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
- [ ] Documentation improvement _(no changes to code)_
- [X] Other _(please specify)_

### CODE REVIEW

**Comment by tomudding** [comment]
Part of a series of enhancements to hopefully make maintainability easier.

**Comment by rinkp** [comment]
How will this work with `SET ROLE` to prevent ownership issues down the road.

I don't think we can add that as a [connection option](https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/configuration.html#pdo-pgsql-pgsql). Do we wish to set the role on every connection or are we going to add this to each migration?

I think the former is preferred, but because it has to happen after creating the connection, will thus always require an extra command.

**Comment by tomudding** [comment]
> How will this work with `SET ROLE` to prevent ownership issues down the road.

Totally forgot about that. Need to think about the best solution. The easiest is adding it to each migration imho as you have to edit them anyway (and I already wanted to add a README for them, so then it can be documented too).

**Comment by tomudding** [comment]
PHPStan fails because it is using the new config on the old main.

**Comment by tomudding** [comment]
**TODO:**
- update internal wiki page on migrations (or remove it)

File: .env.dist [hunk 1]
```diff
@@ -3,6 +3,7 @@ CHECKER_MEMBERSHIP_API_ENDPOINT=https://tue-lookup.test.gewis.nl/user/
 CHECKER_MEMBERSHIP_API_KEY=c2VjcmV0
 CHECKER_MEMBERSHIP_API_MAX_TOTAL_REQUESTS=200
 CHECKER_MEMBERSHIP_API_MAX_MANUAL_REQUESTS=20
+DOCTRINE_ROLE=gewisdb
```
Comment (rinkp) [inline .env.dist:6]: What happens here is that we allow a set of credentials for default and for reportdb: `DOCTRINE_DEFAULT_USER`/`DOCTRINE_REPORT_USER` and `DOCTRINE_DEFAULT_PASSWORD`/`DOCTRINE_REPORT_PASSWORD`

To me it make sense to implement the role on a database-level as well or make the credentials a application-level variable as well. In principle, the credentials determine what roles are available.

Comment (rinkp) [reply .env.dist:6]: What do you think about the solution I propose in https://github.com/tomudding/gewisdb/pull/2

File: module/Application/src/Extensions/Doctrine/Middleware/Driver.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    public function connect(
+        #[SensitiveParameter]
+        array $params,
+    ): ConnectionInterface {
+        $connection = parent::connect($params);
+
+        if ($this->isPgSQL) {
+            $connection->exec('SET ROLE ' . $connection->quote($this->role));
+        }
```
Comment (rinkp) [inline module/Application/src/Extensions/Doctrine/Middleware/Driver.php:33]: I did a proposal for this in tomudding#2

**Review by rinkp — CHANGES_REQUESTED** [review]

File: .env.dist (hunk 1 above)
Comment (tomudding) [reply .env.dist]: Thank you, I have inlined your changes into f80419ffa4270b16b0e7bf2aecd5c4384a279a21.

File: module/Application/src/Extensions/Doctrine/Middleware/Driver.php (hunk 2 above)
Comment (tomudding) [reply module/Application/src/Extensions/Doctrine/Middleware/Driver.php]: Thank you, I have inlined your changes into f80419ffa4270b16b0e7bf2aecd5c4384a279a21.

**Review by rinkp — APPROVED** [review]
Works for me; perhaps we should update the instructions to get the database working initially, but apart from that, migrations seem to be working properly!

**Comment by tomudding** [comment]
Seeding of test data is likely significantly delayed due to ORM issues, see GEWIS/gewisweb#1913 and GEWIS/gewisweb#1918.

**Pull request merged**